tests/test_cancellation.py

```diff
@@ -0,0 +1,66 @@
+import asyncio
+
+import pytest
+
+from aidial_sdk.chat_completion.request import Request as ChatCompletionRequest
+from aidial_sdk.chat_completion.response import (
+    Response as ChatCompletionResponse,
+)
+from aidial_sdk.pydantic_v1 import SecretStr
+from aidial_sdk.utils.streaming import add_heartbeat
+from tests.utils.constants import DUMMY_FASTAPI_REQUEST
+
+
+def create_chat_completion(status: asyncio.Future):
+
+    async def _chat_completion(
+        request: ChatCompletionRequest, response: ChatCompletionResponse
+    ):
+        try:
+            for _ in range(10):
+                await asyncio.sleep(1)
+        except asyncio.CancelledError:
+            status.set_result(True)
+            raise
+        else:
+            status.set_result(False)
+
+    return _chat_completion
+
+
+@pytest.mark.parametrize("with_heartbeat", [True, False])
+async def test_cancellation(with_heartbeat: bool):
```

Comment:
Please, extend this with the test:

1. Run chat completion
2. Inside of chat completion create nested tasks, that run forever 
3. Cancel chat completion
4. Check that nested tasks are also cancelled

Reply:
Fixed, added more tests